src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -128,7 +137,7 @@ private static bool IsValidRootConfigType(ITypeSymbol? type)
                 {
                     ParsableFromStringSpec stringParsableSpec = new(type) { StringParsableTypeKind = specialTypeKind };
 
-                    if (stringParsableSpec.StringParsableTypeKind is not StringParsableTypeKind.AssignFromSectionValue)
+                    if (stringParsableSpec.StringParsableTypeKind != StringParsableTypeKind.AssignFromSectionValue && stringParsableSpec.StringParsableTypeKind != StringParsableTypeKind.Enum)
```

Comment:
I'd re-write this as 

```suggestion
if (specialTypeKind is StringParsableTypeKind.Enum)
{
    _sourceGenSpec.EmitEnumParseMethod = true;
}
else if (specialTypeKind is not StringParsableTypeKind.AssignFromSectionValue)
{
    _sourceGenSpec.PrimitivesForHelperGen.Add(stringParsableSpec);
}
```